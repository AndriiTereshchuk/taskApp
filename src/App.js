import React, { useState } from 'react';
import { ApolloProvider } from '@apollo/client';
import 'bootstrap/dist/css/bootstrap.min.css';

import './App.css';
import apolloClient from './apollo/apollo-client';

import FormContainer from './FormContainer';
import TaskList from './TaskList';
import { useTasksQuery } from "./apollo/autogenerate/hooks";

const MainContainer = () => {
    const { data, refetch: refetchTasks } = useTasksQuery();
    const [task, setTask] = useState(null)
    return(
        <div className="App">
            <FormContainer
                updateTasks={refetchTasks}
                task={task}
            />
            <TaskList
                editTask={setTask}
                tasks={data && data.tasks  || []}
                updateTasks={refetchTasks}
            />
        </div>
    )
}

function App() {
  return (
      <ApolloProvider client={apolloClient}>
          <MainContainer />
      </ApolloProvider>
  );
}

export default App;
