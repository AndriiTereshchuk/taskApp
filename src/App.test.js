import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const { toJSON } = renderer.create(<App />);
  expect(toJSON()).toMatchSnapshot();
});
