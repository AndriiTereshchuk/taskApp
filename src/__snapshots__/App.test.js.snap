// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  <form
    className="Form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="exampleForm.ControlInput1"
      >
        Title
      </label>
      <input
        className="form-control"
        id="exampleForm.ControlInput1"
        onChange={[Function]}
        placeholder="title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="exampleForm.ControlTextarea1"
      >
        Description
      </label>
      <textarea
        className="form-control"
        id="exampleForm.ControlTextarea1"
        onChange={[Function]}
        placeholder="description"
        rows={3}
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
